Dockerfile

```diff
@@ -15,4 +15,5 @@ RUN rm -rf /usr/share/nginx/html/*
 COPY nginx.conf /etc/nginx/
 ARG project
 COPY --from=build /home/node/dist/${project} /usr/share/nginx/html
-CMD ["nginx"]
+# envsubst is used to replace environment variables in env.js when the container is started
+CMD ["/bin/sh", "-c", "envsubst < /usr/share/nginx/html/assets/env.template.js > /usr/share/nginx/html/assets/env.js && exec nginx"]
```

Comment:
Consider creating a `docker-entrypoint.sh` script that is executed here instead. See one of our internal repos like the sidecars for examples.

It should be something like:
```
ENTRYPOINT ["/etc/nginx/docker-entrypoint.sh"]
CMD ["nginx"]
```

Reply:
What happens if this is started without a `CLIENT_ID`? Should there be a `ENV` statement with a default here?

Reply:
I don't think a default CLIENT_ID would make sense. The app will not work unless the client ID is registered with the EHR using the correct launch URL. The localhost client ID will not work when the app is deployed at a different URL. I will add a check in the docker-entrypoint.sh and fail if the env is not configured correctly.

Reply:
The app could be configured to connect to an open FHIR server without SMART launch. In that case, we would want to allow starting the container without a CLIENT_ID defined. However, we do not currently have any way to select a patient when using an open FHIR server. So until/unless we add that feature, I think it still makes sense to fail when the env var is not set. At this stage it is more likely to catch a configuration mistake than to prevent something intentional.

Reply:
Ok, that makes sense.

Reply:
Please change the `CMD` and `ENTRYPOINT` though.